runtime/compiler/control/CompilationThread.cpp

```diff
@@ -6028,6 +6033,48 @@ void *TR::CompilationInfo::compileOnSeparateThread(J9VMThread * vmThread, TR::Il
       if (compErrCode)
          *compErrCode = (TR_CompilationErrorCode)entry->_compErrCode;
 
+      // If a remote sync compilation has been changed to a local sync compilation,
+      // queue a new remote async compilation entry for this method.
+      // - The second compilation is queued only when this is the last thread waiting
+      //   for the compilation of the method.
+      // - If for any reason, the current compilation request is changed to a sync
+      //   compilation from an async compilation, it probably means we can’t do an
+      //   async compilation. Don't queue the second async remote compilation.
+      if (entry->_changedToLocalSyncComp &&
+          (entry->_numThreadsWaiting <= 1) &&
+          !forcedSync &&
+          (startPC && startPC != oldStartPC))
+         {
+         acquireCompMonitor(vmThread);
+
+         void *currOldStartPC = startPCIfAlreadyCompiled(vmThread, details, oldStartPC);
```

Comment:
startPCIfAlreadyCompiled(.., oldStartPC) returns NULL if the method body defined by oldStartPC has not been recompiled. In this case we want to use the existing startPC for addMethodToBeCompiled(). If startPCIfAlreadyCompiled(.., oldStartPC) return something, then it means that a recompilation already happened and we don't have to do anything